charts/coturn/templates/configmap-coturn-conf-template.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "coturn.fullname" . }}
+  labels:
+    {{- include "coturn.selectorLabels" . | nindent 4 }}
+
+data:
+  coturn.conf.template: |
+    ## disable (d)tls control plane; don't permit relaying tcp connections.
+    no-tls
+    no-dtls
+    no-tcp-relay
+
+    ## don't turn on coturn's cli.
+    no-cli
+
+    ## authentication setup
+    # FUTUREWORK: enable support for multiple secrets?
+    zrest
+    static-auth-secret=__COTURN_SECRET__
```

Comment:
Hm, why these identifiers which you replace with `sed` on startup if you could use a variable taken from a kubernetes secret instead? Changes on a new deploy could override the configmap again with these placeholders leading to weird behaviour; as you need to ensure those sed commands always run to completion first.

Probably the existing approach here works; but it's hacky, especially since the secret is available, and you could just use `{{ .Values.secrets.zrestSecret | quote }}` here and entirely avoid the creation of the `charts/coturn/templates/secret.yaml` file

Reply:
This is more cargo-culting from the restund chart. That chart stores the zrest authentication secret in a Kubernetes Secret, loads it into an environment variable in the container command, and then substitutes it into the config file using `envsubst`. The choice of sed here is because the coturn container image I've been working with doesn't contain `envsubst`, and I'm a bit reluctant to add that to the container image, as that would diverge from upstream.

I'd be happy to swap this for a different solution, if you have any suggestions.

Reply:
Oh, something else worth mentioning here is that I don't know how Kubernetes handles access controls or visibility for ConfigMaps and Secrets. Embedding the zrest auth secret into the ConfigMap when it's rendered might mean the secret is more visible within the cluster at runtime than if it were embedded in a Secret.

The brig chart stores various sensitive values (including the TURN zrest auth secret) in a separate Secret object, which attached to the pod separately from the ConfigMap with the main configuration file.